clang-tidy-diff.sh

```diff
@@ -1,13 +1,28 @@
 #!/bin/bash
+
+set -o errexit
+
+# When we are running on travis and *not* part of a pull request we don't
+# have any upstream branch to compare against.
+if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
+  echo "Skipping since not running on travis PR"
+  exit 0
+fi
+
+if [ -n "$TRAVIS_BRANCH" ]; then
+  BRANCH=$TRAVIS_BRANCH
+else
+  BRANCH=origin/master
+fi
+
 CLANG_DIR=$(dirname $(dirname $(which clang-tidy)))
 CLANG_TIDY_DIFF=$CLANG_DIR/share/clang/clang-tidy-diff.py
-MERGE_BASE=$(git merge-base master HEAD)
-TIDY_MSG=$(git diff -U0 $MERGE_BASE | $CLANG_TIDY_DIFF -quiet -p1 2> /dev/null)
+TIDY_MSG=$(git diff -U0 $BRANCH... | $CLANG_TIDY_DIFF -quiet -p1)
 if [ -n "$TIDY_MSG" -a "$TIDY_MSG" != "No relevant changes found." ]
 then
   echo "Fix clang-tidy errors before committing!"
   echo
   # Run clang-tidy once again to show the error
-  git diff -U0 $MERGE_BASE | $CLANG_TIDY_DIFF -quiet -p1 2> /dev/null
```

Comment:
This redirecting stderr is not to show all those skipping messages for headers, like:
```
Skipping /usr/local/google/home/aheejin/binaryen/src/asm2wasm.h. Compile command not found.
```
(Because of how clang-tidy works, errors in header files are only detected when a source file that includes that header was examined.) Should we remove this?

Usual error messages are printed to stdout.